### PR TITLE
Task/48 duration creation

### PR DESCRIPTION
**Pull request by ianbro**
To test, you will have to delete your db and re-migrate (`dotnet ef database update`)

### CODE REVIEW

File: server/budgettracker.common/Models/BudgetDurations/MonthlyDaySpanDuration.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+namespace budgettracker.common.Models.BudgetDurations
+{
+    public class MonthlyDaySpanDuration : BudgetDurationBase
+    {
+        /// <summary>
+        /// <p>
+        /// Specifies the number of days that this budget will span.
+        /// </p>
+        /// </summary>
+        public int NumDays { get; set; }
```
Comment (BenjaminGroseclose) [inline server/budgettracker.common/Models/BudgetDurations/MonthlyDaySpanDuration.cs]: NumberDays we don't need to abbreviate here.

File: server/budgettracker.data/BudgetTrackerContext.cs [hunk 2]
```diff
@@ -8,10 +8,11 @@ public class BudgetTrackerContext : DbContext
         public BudgetTrackerContext(DbContextOptions options)
             :base(options)
         {
-            
+
         }
 
         public DbSet<UserModel> Users { get; set; }
         public DbSet<BudgetModel> Budgets { get; set; }
+        public DbSet<BudgetDurationModel> BudgetDurations { get; set; }
```
Comment (BenjaminGroseclose) [inline server/budgettracker.data/BudgetTrackerContext.cs]: Is this the budget periods why do we need a separate table for BudgetDuration

File: server/budgettracker.business/Api/Contracts/BudgetApi/BudgetDurations/MonthlyDaySpanDurationContract.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+namespace budgettracker.business.Api.Contracts.BudgetApi.BudgetDurations
+{
+    public class MonthlyDaySpanDurationContract : BudgetDurationBaseContract
```
[the remark this answers is not in the document]
Comment (ianbro) [reply server/budgettracker.business/Api/Contracts/BudgetApi/BudgetDurations/MonthlyDaySpanDurationContract.cs]: Crap... I need to test this version of the duration. It is a contract, I just forgot to do make it jsonizable. I tested the bookended one but not this one.

File: server/budgettracker.business/Api/Contracts/BudgetApi/CreateBudgetRequestContract.cs [hunk 4]
```diff
@@ -13,15 +16,34 @@ public class CreateBudgetRequestContract : IApiContract {
...
-        /// </summary>    
+        /// </summary>
         [JsonProperty("duration")]
-        public int Duration { get; set; }
+        public JObject DurationTemp { get; set; }
+
+        public BudgetDurationBaseContract Duration {
+            get {
+                string durationSerialized = JsonConvert.SerializeObject(DurationTemp);
+                if (DurationTemp.ContainsKey("start-day-of-month") &&
```
[the remark this answers is not in the document]
Comment (ianbro) [reply server/budgettracker.business/Api/Contracts/BudgetApi/CreateBudgetRequestContract.cs]: What do you mean by "using strings like this"?

File: server/budgettracker.data/BudgetTrackerContext.cs (hunk 2 above)
Comment (ianbro) [reply server/budgettracker.data/BudgetTrackerContext.cs]: Lol maybe I should explain better what this PR is doing. I'll explain next time we get a chance in person.

File: server/budgettracker.business/Api/Contracts/BudgetApi/CreateBudgetRequestContract.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (ianbro) [reply server/budgettracker.business/Api/Contracts/BudgetApi/CreateBudgetRequestContract.cs]: @BenjaminGroseclose You approved the changes. Is this resolved then?

**Pull request merged**